samila/genimage.py

```diff
@@ -80,10 +79,12 @@ def generate(
         range1 = list(float_range(self.start, self.stop, self.step))
         range2 = list(float_range(self.start, self.stop, self.step))
         range_prod = list(itertools.product(range1, range2))
-        for item in range_prod:
-            random.seed(self.seed)
-            self.data1.append(self.function1(item[0], item[1]).real)
-            self.data2.append(self.function2(item[0], item[1]).real)
+        calc_exception = False
+        for point in range_prod:
+            if fill_data(self, point):
```

Comment:
I think it should be:
```python3
not fill_data(self, point)
```

Reply:
It's odd that all of our tests were passed :)